internal/server/models/access_key.go

```diff
@@ -46,6 +46,7 @@ func (ak *AccessKey) ToAPI() *api.AccessKey {
 		ID:                ak.ID,
 		Name:              ak.Name,
 		Created:           api.Time(ak.CreatedAt),
+		LastUsed:          api.Time(ak.UpdatedAt),
```

Comment:
On first read this seemed wrong, but I guess it works in some cases because of the extension deadline. If the key has an extension deadline then we'll update it on use, so `UpdatedAt` will be updated.

However I believe there are 2 cases where that may not work:
* if the key doesn't have an extension deadline then we don't update it, which means `LastUsed` will be wrong (it'll show the created time always)
* if we get to the period of time where the extension deadline has reached the expiration, so it can't be extended anymore. This could potentially be a long period of time if the extension deadline is large. I think today that may not be an issue because I'm not sure we're handling this correctly, but once we fix that bug it'll be a problem here.

I think we may need a new field for `LastUsed` to accurately show the used time.

Reply:
This is interesting, however I don't think we _currently_ would hit either of those cases.
1. Right now we _require_ the user to include the `extensionDeadline` if you're creating the key through the API, otherwise you'll get a 400. This could be abused by us if someone creates an access key outside of the API, so it feels a little bit brittle.
2. Once the expiry is hit, the key won't work anymore. The extensionDeadline is ignored, so `UpdatedAt` will never get updated, but I think this is the correct value (i.e. the last time the access key was successfully used). I can see an argument for changing it if we want to record attempted use instead of actual use.

For now, I'll leave some comments in the code.

Reply:
There are ways  to create an access key without an extension deadline I think.

`Signup` and `Server.loadConfig` both create an access key without an extension deadline, it can't be extended.

For 2, I don't think I explained it well. The issue is not that the key is expired, it's that the key can't be extended anymore. For example:

1. Extension deadline set to Dec 1st 12:00, with an extension value of 30 days
2. After Nov 1st, we shouldn't be making the expiry later than Dec 1st 12:00, because then the expiry would be past the extension deadline.

Today that's what we do, which means we are forced to look at two different values to determine if the key is valid or not. See #3118

During that period of time (nov 1st to dec 1st) the key won't be extended anymore, so the lastSeenAt will be wrong. In this example it's only 30 days, but it could be much longer if the extension value is longer.